sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -723,6 +746,19 @@ impl SocRegistersImpl {
                     Some(self.timer.schedule_poll_in(Self::FW_READ_TICKS));
             }
         }
+
+        if self
+            .timer
+            .fired(&mut self.op_idevid_csr_read_complete_action)
+        {
+            // Download the Idevid CSR from the mailbox.
+            (self.download_idevid_csr_cb)(&mut self.mailbox, &self.log_dir);
+
+            // Clear the Idevid CSR requested bit.
+            self.cptra_dbg_manuf_service_reg
```

Comment:
I don't think the verilog does this. Shouldn't this be done by the callback (maybe by passing in this register as an argument)?

Reply:
Fixed.